util.go

```diff
@@ -104,3 +104,11 @@ func setTLSContext(ctx context.Context, caBundle []byte) context.Context {
 	tlsConf := &http.Client{Transport: tr}
 	return context.WithValue(ctx, oauth2.HTTPClient, tlsConf)
 }
+
+func doRequest(ctx context.Context, req *http.Request) (*http.Response, error) {
+	client := http.DefaultClient
+	if c, ok := ctx.Value(oauth2.HTTPClient).(*http.Client); ok {
+		client = c
+	}
+	return client.Do(req.WithContext(ctx))
+}
```

Comment:
In the spec, I see that the server may return a 503 response, in which case the client can retry the request (granted, this holds for all 503 requests). In this case, do you know if the requests are retried internally?

Of course, if we expect that the retries will be handled by upper layers, this is a moot question.

Reply:
No retries happen for now.
In this case, the revocation and subsequent logout will fail.
I will add a TODO here, sounds ok?

Reply:
Sure, that's ok.